packages/components/molecules/f-restaurant-card/src/components/subcomponents/DeliveryTimeMeta/_tests/DeliveryTimeMeta.test.js

```diff
@@ -0,0 +1,193 @@
+import { shallowMount, mount, config } from '@vue/test-utils';
+import sut from '../DeliveryTimeMeta.vue';
+
+
+describe('DeliveryTimeMeta', () => {
+    it('should be defined', () => {
+        // arrange
+        const propsData = {
+            eta: '20 - 25 min'
+        };
+
+        // act
+        const wrapper = shallowMount(sut, { propsData });
+
+        // assert
+        expect(wrapper.exists()).toBe(true);
+    });
+    it('should show eta text when data is present', () => {
+        // arrange
+        const propsData = {
+            eta: 'foo'
+        };
+
+        // act
+        const wrapper = mount(sut, { propsData });
+
+        const visibleText = wrapper.find('[data-test-id="icon-text-visible-text"]').text();
+
+        // assert
+        expect(visibleText).toBe('foo');
+    });
+
+    it('should fallback to distance when eta data is missing', () => {
+        // arrange
+        const propsData = {
+            distance: 'bar'
+        };
+
+        // act
+        const wrapper = mount(sut, { propsData });
+
+        const visibleText = wrapper.find('[data-test-id="icon-text-visible-text"]').text();
+
+        // assert
+        expect(visibleText).toBe('bar');
+    });
+
+    it('should fallback to address when eta and distance data is missing', () => {
+        // arrange
+        const propsData = {
+            address: 'baz'
+        };
+
+        // act
+        const wrapper = mount(sut, { propsData });
+
+        const visibleText = wrapper.find('[data-test-id="icon-text-visible-text"]').text();
+
+        // assert
+        expect(visibleText).toBe('baz');
+    });
+
+    it('should only show 1 Icon Text', () => {
+        // arrange
+        const propsData = {
+            eta: 'foo',
+            distance: 'bar',
+            address: 'baz'
+        };
+
+        // act
+        const wrapper = mount(sut, { propsData });
+
+        const numberOfIconTextElements = wrapper.findAll('[data-test-id="icon-text-visible-text"]').length;
+
+        // assert
+        expect(numberOfIconTextElements).toBe(1);
+    });
+
+    describe('Tile view', () => {
+        it('should hide eta icon but show text', () => {
+            // arrange
+            const propsData = {
+                eta: 'foo'
+            };
+
+            // act
+            const wrapper = mount(sut, { propsData });
+
+            const etaIconExists = wrapper.findAll('[data-test-id="eta-icon-text"] [data-test-id="icon-text-icon"]').exists();
+            const etaTextExists = wrapper.findAll('[data-test-id="eta-icon-text"] [data-test-id="icon-text-visible-text"]').exists();
+
+            // assert
+            expect(etaIconExists).toBe(false);
+            expect(etaTextExists).toBe(true);
```

Comment:
I'm in two minds about this. Its very tightly coupled to the implementation of icon text. You could always use shallowMount here - that will render out something like `<icon-text-stub show-icon="false" />` - you could then test that the correct attribute is added instead of relying on the icon text implementation (which could change at any time)

Try using shallowMount and logging out `wrapper.html()` and hopefully you'll see what I mean?

Reply:
I see what you're saying, I think I have a nice solution for this, I'll see if it works 🤔